cityofaustin/joplin
# 4257 user importer

# Description
Getting users into importer was absolute madness

- [ ] We can now import users as page owners.
- [ ] But! You must have an authentication token in order to query users. So now our import functionality requires you to pass in a JWT token.
<img width="513" alt="Screen Shot 2020-04-12 at 11 19 27 PM" src="https://user-images.githubusercontent.com/14187277/79091744-1dbd8900-7d14-11ea-9984-ed7551594b82.png">
<img width="479" alt="get_graphql_auth_token" src="https://user-images.githubusercontent.com/14187277/79091794-49d90a00-7d14-11ea-8c79-2d06551428a7.png">

- [ ] README/authorization_graphql.md has instructions for getting an API token.
- [ ] `pytest` will fetch a token automatically for you. However, you must add the API_TEST_USER_PASSWORD into your .env file (and Pycharm Run configuration). More info is in README/environment_variables.md. The API_TEST_USER_PASSWORD is in 1password and I'll slack it.
- [ ] Adding security necessarily means that there are extra dumb hoops to jump through. It's unavoidable to an extent, but it's a good thing for our users. That being said, I don't totally love requiring an extra env var to get pytest to work. I made an issue to address this another time: https://github.com/cityofaustin/techstack/issues/4268

What else did I do?
- [ ] We've now consolidated our create_fixture() function into 1 place for all page_types.
- [ ] I had to drop and reload the "pytest" branch database. I actually didn't have to. But I did anyway and now we all have to live with the consequences. Since the revision_ids have changed, all of your branches will fail on pytest until this branch is merged in. I apologize. However, the pytest branch now has a complete "dummy" datadump of all the data it needs so hopefully this will never happen again.
- [ ] Our PR vars automatically set "LOAD_DATA": "fixtures" and "V3_WIP": True. So no more branch configs that we always forget to set.
- [ ] ISSTAGING, ISPRODUCTION, etc. are now snake cased IS_STAGING, IS_PRODUCTION. That was just driving me crazy.
- [ ] Set a SUPERADMIN_USER_PASSWORD in your .env to set the password for admin@austintexas.io. We've got a UserFactory to load in test users instead of a datadump. It's more secure than having our fixture show the hashed password. For PRs, the SUPERADMIN_USER_PASSWORD is set in circleci, also it'll be on slack.








# Types of changes



# How can this be tested?



 

http://joplin-pr-4257-user-importer.herokuapp.com/admin/login/
1. Import a page.
2. Check that it imported the owner of the page.

# Checklist:
